### PR TITLE
feat(rocprofiler-sdk): Emit ROCTx ranges from record_function in PyTorch

### PR DESCRIPTION
**Pull request by EvanPhoukong**
Corresponding [PyTorch PR](https://github.com/ROCm/pytorch/pull/3483), explained simply in the "Standard Pipeline" section

## Motivation

PyTorch users can label workload regions with `record_function`, and those labels appear in TensorBoard traces. The same labels are not automatically available in rocprofv3 or rocprof-sys, forcing users to duplicate their instrumentation with explicit ROCTx calls.

This change allows existing `record_function` labels to appear as ROCTx marker regions without modifying application code.

## Technical Details

- Added `--pytorch-trace` to rocprofv3 and rocprof-sys.
- The option sets `TORCH_PROFILER_EMIT_ROCTX=1` for the target process.
- A corresponding PyTorch change uses the existing `roctx` Python module to emit `record_function` ranges when this setting is enabled.
- rocprofv3 automatically enables marker tracing when `--pytorch-trace` is used.
- rocprof-sys consumes the emitted ranges through its existing `marker_api` support.
- rocprofv3 attach mode is rejected because marker emission must be enabled before PyTorch starts.
- Added CLI tests, configuration schema updates, and user documentation.

**Standard Pipeline**

The normal --pytorch-trace pipeline is as follow. **This PR focuses on implementing this new parameter.**

```
rocprofv3 --pytorch-trace -- python app.py
  → launcher sets TORCH_PROFILER_EMIT_ROCTX=1
  → Python starts and imports PyTorch
  → PyTorch caches “ROCTx enabled”
```

**The below steps of the pipeline are implemented in the PyTorch PR: https://github.com/ROCm/pytorch/pull/3483**

When the application reaches:
```
with record_function("optimizer_step"):
    optimizer.step()
```

the calls are:

```
record_function.__enter__()
  → _record_function_enter_new(...)       # normal PyTorch profiler record
  → _get_roctx_module()                   # import once, then cached
  → roctx.rangePush("optimizer_step")
    → libpyroctx.roctxRangePush(...)
    → native roctxRangePushA(...)
    → profiler callback records start time and label

optimizer.step()                           # application/GPU work happens

record_function.__exit__()
  → _record_function_exit(...)            # finish normal PyTorch record
  → roctx.rangePop()
    → libpyroctx.roctxRangePop()
    → native roctxRangePop()
    → profiler callback records end time
    → completed range is written to profiler buffer
```

## Issue Tracking

JIRA ID: AIPROFSDK-128

## Test Plan

Run the integration tests and successfully profile a real PyTorch Application.

1. Run the targeted rocprofv3 option and environment-propagation tests.
2. Build rocprof-sys and run its CLI help and environment-propagation tests.
3. Profile a real ROCm PyTorch workload containing:

   ```python
   with record_function("optimizer_step"):
       optimizer.step()
   ```

4. Verify that both profiler outputs contain an `optimizer_step` marker region with GPU work inside it.

## Test Result

- All targeted rocprofv3 integration tests passed.
- rocprof-sys built successfully and its targeted tests passed.
- Validated on an AMD Instinct MI300X with PyTorch 2.10.0 and ROCm 7.1.
- rocprofv3 and rocprof-sys both captured the `optimizer_step` range.
- Each captured range enclosed the expected optimizer GPU kernel.

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/rocm-systems/blob/develop/CONTRIBUTING.md.

### CODE REVIEW

**Comment by sputhala-amd** [comment]
@EvanPhoukong - I just got notified by Git and wanted to mention one concern. The rocshmem PR with the fix possibly didnt make it to rocm 10, hence users using that build might still need the fix?
